packages/amplify-provider-awscloudformation/src/iterative-deployment/deployment-manager.ts

```diff
@@ -243,10 +257,11 @@ export class DeploymentManager {
       });
     });
 
-    await Promise.all(waiters);
     try {
+      await Promise.all(waiters);
```

Comment:
Can this be kept as it was initally. If the promise gets rejected, then it will fail the whole deployment.